src/core/Ekubo.sol

```diff
@@ -0,0 +1,453 @@
+                unchecked {
+                    state.buy.amount += settledBuyAmount.asCredit(state.buy.token);
+                }
+            }
+        }
+
+        // `data` has been consumed. All that remains is to settle out the net result of all the
+        // swaps. Any credit in any token other than `state.buy.token` will be swept to
+        // Settler. `state.buy.token` will be sent to `recipient`.
+        {
+            (IERC20 globalSellToken, uint256 globalSellAmount) = (state.globalSell.token, state.globalSell.amount);
+            uint256 globalBuyAmount =
+                Take.take(state, notes, uint32(IEkuboCore.withdraw.selector), recipient, minBuyAmount);
+            if (feeOnTransfer) {
+                // We've already transferred the sell token to the vault and
+                // `settle`'d. `globalSellAmount` is the verbatim credit in that token stored by the
+                // vault. We only need to handle the case of incomplete filling.
+                if (globalSellAmount != 0) {
+                    Take._callSelector(
+                        uint32(IEkuboCore.withdraw.selector),
+                        globalSellToken,
+                        (payer == address(this))? address(this) : _msgSender(),
+                        globalSellAmount
+                    );
+                }
+            } else {
+                // While `notes` records a credit value, the vault actually records a debt for the
+                // global sell token. We recover the exact amount of that debt and then pay it.
+                // `globalSellAmount` is _usually_ zero, but if it isn't it represents a partial
+                // fill. This subtraction recovers the actual debt recorded in the vault.
+                uint256 debt;
+                unchecked {
+                    debt = state.globalSellAmount - globalSellAmount;
+                }
+                if (debt == 0) {
+                    revert ZeroSellAmount(globalSellToken);
+                }
+                _ekuboPay(globalSellToken, payer, debt, permit, isForwarded, sig);
+            }
+
+            // return abi.encode(globalBuyAmount);
+            bytes memory returndata;
+            assembly ("memory-safe") {
+                returndata := mload(0x40)
+                mstore(returndata, 0x60)
+                mstore(add(0x20, returndata), 0x20)
+                mstore(add(0x40, returndata), 0x20)
+                mstore(add(0x60, returndata), globalBuyAmount)
+                mstore(0x40, add(0x80, returndata))
+            }
+            return returndata;
+        }
+    }
+
+    function payCallback(bytes calldata data) private returns (bytes memory) {
+        IERC20 sellToken;
+        uint256 requestedAmount;
+        uint256 sellAmount;
+        
+        ISignatureTransfer.PermitTransferFrom calldata permit;
+        bool isForwarded;
+        bytes calldata sig;
+
+        
+        assembly ("memory-safe") {
+            
+        }
```

Comment:
?

Reply:
last minute push before going off 🤦‍♂️, moved the code into the next assembly section and did not remove that